feat: add slack notification on failed integration tests

hanshal101 opened the pull request:
Fixes #332 
Closes #332 

### Description
This PR adds the feature to give alerts on slack when the integration tests fails.

### TODO
@anfredette , we need to add a Github secret named `SLACK_WEBHOOK_URL` to enable the alerts.

hanshal101 committed:
feat: add slack notification on failed integration tests

Signed-off-by: Hanshal Mehta <122217807+hanshal101@users.noreply.github.com>

dave-tucker commented on .github/workflows/integration_test.yml:
```diff
@@ -80,3 +80,10 @@ jobs:
+
+      - name: Notify Slack on Test Failure
+        if: failure()
+        uses: rtCamp/action-slack-notify@v2
```
It might be better to use the action supported by Slack themselves.
There's a fancy example here: https://github.com/slackapi/slack-github-action/blob/main/example-workflows/Technique_3_Slack_Incoming_Webhook/blocks.yml
And a much simpler one here: https://github.com/slackapi/slack-github-action/blob/main/example-workflows/Technique_3_Slack_Incoming_Webhook/text.yml

dave-tucker commented on .github/workflows/integration_test.yml:
```diff
@@ -80,3 +80,10 @@ jobs:
+        uses: rtCamp/action-slack-notify@v2
+        with:
+          webhook_url: ${{ secrets.SLACK_WEBHOOK_URL }}
+          message: "Nightly Kubernetes BPFMan Operator integration tests failed! Go version: ${{ matrix.go }} and OCI runtime: ${{ matrix.oci_bin }}"
```
```suggestion
          message: "Nightly Kubernetes bpfman Operator integration tests failed! Go version: ${{ matrix.go }} and OCI runtime: ${{ matrix.oci_bin }}"
```

dave-tucker reviewed (changes requested):
Thanks for the contribution. I left some comments, but reading up on this some more it seems like webhooks are ~being deprecated~ no longer recommended for this usecase + we don't have the ability to add them easily on the K8s slack instance. 

We can certainly request a webhook from the admins, but it might make more sense to do use the Github integration like this: https://github.com/integrations/slack?tab=readme-ov-file#workflow-notification-filters

That's more flexible since we can subscribe/unsubscribe to notifications in the channel whenever we like.

hanshal101 committed:
feat: add slack notification on failed integration tests

Signed-off-by: Hanshal Mehta <122217807+hanshal101@users.noreply.github.com>

hanshal101 commented:
Thanks @dave-tucker for the review, I have updated the CI to the official `slackapi`, I have added a simple string for the message, let me know if we need more detailed one or not.
